connect/runtime/src/test/java/org/apache/kafka/connect/util/clusters/EmbeddedConnectCluster.java

```diff
@@ -126,11 +154,46 @@ public void startConnect() {
         putIfAbsent(workerProps, KEY_CONVERTER_CLASS_CONFIG, "org.apache.kafka.connect.storage.StringConverter");
         putIfAbsent(workerProps, VALUE_CONVERTER_CLASS_CONFIG, "org.apache.kafka.connect.storage.StringConverter");
 
-        for (int i = 0; i < connectCluster.length; i++) {
-            connectCluster[i] = new ConnectDistributed().startConnect(workerProps);
+        for (int i = 0; i < numInitialWorkers; i++) {
+            connectCluster.add(new ConnectDistributed().startConnect(workerProps));
         }
     }
 
+    /**
+     * Get the workers that are up and running.
+     *
+     * @return the list of online workers
+     */
+    public Collection<String> activeWorkers() {
+        ObjectMapper mapper = new ObjectMapper();
+        return connectCluster.stream()
+                .map(Connect::restUrl)
+                .map(URI::toString)
+                .map(url -> {
+                    try {
+                        mapper.readerFor(ServerInfo.class).readValue(executeGet(url));
+                        return url;
+                    } catch (IOException e) {
+                        // Worker failed to respond. Consider it's offline
+                        return null;
```

Comment:
I'm curious: what happens if the worker's request thread is blocked forever? IIRC, this request should ultimately time out, but might this cause a flaky tests because it times out after too long?

Reply:
I simply follow here what is done elsewhere in the harness, with calls directly to the rest interface of the worker. Is this a general concern regarding the rest calls? Usually you'd use `activeWorkers` within an assert similar to the `assertWorkersUp` I have in the example.

Reply:
A maybe slightly interesting aspect is that this method returns now a ternary value, because it's up to you how you want to interpret `Optional.empty` (meaning failure to get a definite answer).  I thought it could work with negation of assertions.